<div style="height:1123px; font-family: 'Times New Roman', Times, serif; font-size:14px; text-align:center;">

## **LAPORAN PRAKTIKUM**

## **ALGORITMA & PEMOGRAMAN**

## **“OPERATOR PADA PYHTON”**

## **Disusun Oleh :**

## **AGUNG FIRMANSYAH**

## **2611533011**

## **Dosen Pengampu:**

## **Wahyudi.DR.. S.T.M.T**

## **Asisten Praktikum:**

## **SHERLY SUKMADIRA**

<br>
<img src="images/logo_unand.png" width="200">

<br>

## **FAKULTAS TEKNOLOGI INFORMASI**

## **DEPARTEMEN INFORMATIKA**

## **UNIVERSITAS ANDALAS**

## **2026**

</div>

---
<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">
<h1 align="center"><b>KATA PENGANTAR</b></h1>

Puji syukur penulis panjatkan atas kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan berkatnya, sehingga penulis dapat menyelesaikan laporan praktikum tentang Operator dalam Python dan variabel yang telah dilaksanakan sebagai pertemuan ke-3 untuk mata kuliah Praktikum Struktur Data. Praktikum ini berfokus pada Operator dalam Python yang digunakan untuk menentukan jenis nilai program Aritmatika, Assignment, Bitwise, Keanggotaan, Identitas, Perbandingan, Logika

Melalui praktikum ini, hasil yang diharapkan adalah penulis dapat memahami berbagai jenis Operator dalam Python dan cara menggunakan variabel dalam Python serta mampu menerapkannya dalam pembuatan program sederhana. Operator dalam Python adalah jenis data yang digunakan dalam program, seperti angka, teks, atau nilai benar dan salah. Operator dalam Python membantu komputer mengetahui cara mengolah data tersebut. Variabel adalah tempat untuk menyimpan data dalam program. Data yang disimpan dalam variabel bisa digunakan kembali atau diubah sesuai kebutuhan.

Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan di masa mendatang.

<div align="right" style="margin-left:-30px;">
Padang, 2026
<br>
<br>
<br>
<br>
<br>
<br>
Agung Firmansyah
<br>
2611533011
</div>
</div>

---

<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">
<h1 align="center"><b>BAB I <br> PENDAHULUAN</b></h1>

## 1.1 Latar Belakang
<p style="text-indent: 40px; text-align: justify; ">Dalam pemrograman Python, operator merupakan salah satu dasar yang penting untuk dipahami karena digunakan untuk melakukan berbagai proses terhadap data. Operator memungkinkan program untuk melakukan perhitungan, membandingkan nilai, memberikan atau mengubah nilai pada variabel, mengolah kondisi, serta melakukan operasi tertentu terhadap data. Penggunaan operator yang tepat dapat membantu program menghasilkan proses dan keputusan sesuai dengan kebutuhan. Python memiliki berbagai jenis operator, seperti operator aritmatika, assignment, perbandingan, logika, keanggotaan, identitas, dan bitwise. Setiap operator memiliki fungsi dan cara penggunaan yang berbeda. Dengan memahami jenis dan fungsi dari masing-masing operator, pengguna dapat membuat program yang lebih terstruktur serta mampu mengolah data dengan lebih efektif. Oleh karena itu, praktikum mengenai operator dalam Python dilakukan untuk memahami penggunaan berbagai jenis operator secara langsung melalui program. Melalui praktikum ini, diharapkan dapat mengetahui cara kerja operator serta menerapkannya dalam menyelesaikan permasalahan sederhana menggunakan bahasa pemrograman Python.
</p>

## 1.2 Tujuan
1. Mengetahui pengertian dan fungsi Operator Python.
2. Memahami jenis-jenis Operator dalam Python.
3. Mengetahui cara membuat dan menggunakan variabel.
4. Mampu menggunakan Operator Python dalam membuat program sederhana.
5. Memahami cara menyimpan dan mengolah data dalam program
## 1.3 Manfaat
1. Mahasiswa memahami dasar-dasar pemrograman dengan lebih mudah.
2. Mahasiswa meningkatkan kemampuan berpikir logis dalam menyelesaikan masalah.
3. Mahasiswa mampu membuat program sederhana menggunakan Python.
4. Mahasiswa memahami cara menyimpan dan mengolah data dalam program.
5. Mahasiswa menjadi bekal untuk mempelajari pemrograman lebih lanjut.
</div>

---
<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
<h1 align="center"><b>BAB II<br>PEMBAHASAN </b> </h1>

## 2.1 Dasar Teori
Pengenalan tentang operator dalam python


1. Operator Aritmatika<br>
Operator aritmatika adalah operator yang digunakan untuk melakukan operasi matematika. Operator ini digunakan dalam berbagai proses perhitungan, seperti penjumlahan, pengurangan, perkalian, dan pembagian.

2. Operator Assignment<br>
Operator assignment atau operator penugasan digunakan untuk memberikan suatu nilai kepada variabel. Operator ini juga dapat digunakan untuk mengubah nilai yang sudah tersimpan pada sebuah variabel.

3. Operator Bitwise<br>
Operator bitwise merupakan operator yang digunakan untuk melakukan operasi pada tingkat bit dari suatu bilangan. Bilangan akan diproses berdasarkan bentuk binernya, yaitu menggunakan angka 0 dan 1.

4. Operator Keanggotaan<br>
Operator keanggotaan atau membership operator digunakan untuk memeriksa apakah suatu nilai terdapat di dalam sebuah kumpulan data.

5. Operator Identitas<br>
Operator identitas atau identity operator digunakan untuk memeriksa apakah dua variabel mengacu pada objek yang sama.

6. Operator Perbandingan<br>
Operator perbandingan digunakan untuk membandingkan dua nilai atau lebih. Hasil dari perbandingan biasanya berupa nilai True atau False, yang menunjukkan apakah suatu kondisi benar atau salah.

7. Operator Logika<br>
Operator logika digunakan untuk menggabungkan atau memanipulasi beberapa kondisi. Operator ini sering digunakan dalam pengambilan keputusan pada suatu program.
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

## 2.2 Kode Program

### 2.2.1 Program "Operator Aritmatika_2611533011"
</div>

In [1]:

# buat program untuk operator aritmatika dalam python
# nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# program ini menggunakan fungsi input()
# nilai yang di masukkan akan di konversi menjadi tipe data integer

angka1_3011 = int(input("input angka-1:"))
angka2_3011 = int(input("input angka-2:"))

# PENJUMLAHAN
hasil_3011 = angka1_3011 + angka2_3011
print("\nOperator Penjumlahan")
print("Hasil =", hasil_3011)

# PENGURANGAN
hasil_3011 = angka1_3011 - angka2_3011
print("\nOperator Pengurangan")
print("Hasil =", hasil_3011)

# PERKALIAN
hasil_3011 = angka1_3011 * angka2_3011
print("\nOperator Perkalian")
print("Hasil =", hasil_3011)

# PEMBAGIAN
if angka2_3011 != 0:
    hasil_3011 = angka1_3011 / angka2_3011
    print("\nOperator Pembagian")
    print("Hasil =", hasil_3011)

    hasil_3011 = angka1_3011 // angka2_3011
    print("\nOperator Pembagian Bulat")
    print("Hasil =", hasil_3011)

    hasil_3011 = angka1_3011 % angka2_3011
    print("\nOperator Sisa Bagi")
    print("Hasil =", hasil_3011)
else:
    print("angka kedua tidak boleh bernilai 0.")

# PANGKAT
hasil_3011 = angka1_3011 ** angka2_3011
print("\n Operator Pangkat")
print("Hasil=", hasil_3011)


Operator Penjumlahan
Hasil = 15

Operator Pengurangan
Hasil = -1

Operator Perkalian
Hasil = 56

Operator Pembagian
Hasil = 0.875

Operator Pembagian Bulat
Hasil = 0

Operator Sisa Bagi
Hasil = 7

 Operator Pangkat
Hasil= 5764801


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

Penjelasan:
1. Penjumlahan<br>
Angka pertama yaitu 7 dijumlahkan dengan angka kedua yaitu 8. Hasilnya adalah 15.<br>
7 + 8 = 15
2. Pengurangan<br>
Angka pertama dikurangi dengan angka kedua. Karena angka pertama lebih kecil dari angka kedua, hasilnya menjadi -1.
7 - 8 = -1
3. Perkalian<br>
Angka 7 dikalikan dengan angka 8 sehingga menghasilkan 56.<br>
7 × 8 = 56
4. Pembagian<br>
Angka 7 dibagi dengan angka 8. Hasilnya berupa bilangan desimal, yaitu 0.875.<br>
7 ÷ 8 = 0.875
5. Pembagian Bulat<br>
Pada pembagian bulat, hasil pembagian 7 dengan 8 hanya mengambil bagian bilangan bulatnya. Karena hasil pembagian adalah 0.875, maka hasil pembagian bulatnya adalah 0.<br>
7 // 8 = 0
6. Sisa Bagi<br>
Operasi sisa bagi digunakan untuk mengetahui sisa dari pembagian 7 dengan 8. Karena 7 masih lebih kecil dari 8, maka seluruh angka 7 menjadi sisa pembagian.<br>
7 % 8 = 7<br>
Jadi, hasilnya adalah 7.
7. Pangkat<br>
Pada operasi pangkat, angka pertama yaitu 7 menjadi bilangan dasar dan angka kedua yaitu 8 menjadi pangkat.
7⁸ = 5.764.801<br>
Sehingga hasil yang ditampilkan adalah 5764801.
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

### 2.2.2 Program "Operator Assignment_2611533011"
</div>

In [2]:
# nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# program ini menggunakan fungsi input()
# nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# program operator assignment dalam python

angka1_3011 = int(input("input angka-1: "))
angka2_3011 = int(input("input angka-1: "))

print("\n nilai awal angka1 =", angka1_3011)
print("nilai angka2 =", angka2_3011)

# assignment biasa 
hasil_3011 = angka1_3011
print("\n assingmnet biasa (=)")
print("Hasil =", hasil_3011)

# assignment penambahan
hasil_3011 = angka1_3011
hasil_3011 += angka2_3011
print("\n assignment penambahan (+=)")
print("Hasil =", hasil_3011)

# assignment perkalian
hasil_3011 = angka1_3011
hasil_3011 *= angka2_3011
print("\n assignment perkalian (*=)")
print("Hasil =", hasil_3011)

# assignment pembagian, pembagian bulat, dan sisa bagi 
if angka2_3011 != 0:
    hasil_3011 = angka1_3011
    hasil_3011 /= angka2_3011
    print("\n assignment pembagian (/=)")
    print("Hasil =", hasil_3011)

    #operator tambahan
    hasil_3011 = angka1_3011
    hasil_3011 //= angka2_3011
    print("\n assignment pembagian bulat (//=)")
    print("Hasil =", hasil_3011)

    hasil_3011 = angka1_3011
    hasil_3011 %= angka2_3011
    print("\n assignment sisa bagi (%=)")
    print("Hasil =", hasil_3011)

else:
    print("\n pembagian tidak dapat dilakukan")
    print("angka kedua tidak boleh 0")


 nilai awal angka1 = 8
nilai angka2 = 5

 assingmnet biasa (=)
Hasil = 8

 assignment penambahan (+=)
Hasil = 13

 assignment perkalian (*=)
Hasil = 40

 assignment pembagian (/=)
Hasil = 1.6

 assignment pembagian bulat (//=)
Hasil = 1

 assignment sisa bagi (%=)
Hasil = 3


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Nilai Awal<br>
Program menampilkan nilai awal dari kedua variabel. angka1 memiliki nilai 8, sedangkan angka2 memiliki nilai 5.<br>
2. Assignment Biasa (`=`)<br>
Assignment biasa digunakan untuk memberikan nilai dari angka1 kepada variabel hasil. Karena nilai angka1 adalah 8, maka hasil yang ditampilkan adalah 8.<br>
Hasil = 8
3. Assignment Penambahan (`+=`)<br>
Operator += digunakan untuk menambahkan nilai angka2 ke nilai yang sudah terdapat pada hasil.
8 + 5 = 13<br>
Jadi, hasil yang ditampilkan adalah 13.
4. Assignment Perkalian (`*=`)<br>
Operator *= digunakan untuk mengalikan nilai hasil dengan nilai angka2.
8 × 5 = 40<br>
Jadi, hasil yang ditampilkan adalah 40.
5. Assignment Pembagian (`/=`)<br>
Operator /= digunakan untuk membagi nilai hasil dengan angka2.
8 ÷ 5 = 1.6<br>
Karena pembagian biasa dapat menghasilkan bilangan desimal, maka output yang diperoleh adalah 1.6.
6. Assignment Pembagian Bulat (`//=`)<br>
Operator //= digunakan untuk melakukan pembagian dan mengambil hasil bilangan bulatnya.
8 ÷ 5 = 1.6<br>
Bagian bilangan bulat dari 1.6 adalah 1, sehingga hasil yang ditampilkan adalah 1.
7. Assignment Sisa Bagi (`%=`)<br>
Operator %= digunakan untuk mendapatkan sisa dari pembagian antara angka1 dan angka2.
8 ÷ 5 = 1 sisa 3<br>
Jadi, hasil sisa bagi yang ditampilkan adalah 3.
### 2.2.3 Program "Operator Bitwise_2611533011"
</div>

In [5]:
# nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# program ini menggunakan fungsi input()


print("\n========================================")
print("3. Operator Bitwise")
print("==========================================")

angka1_3011 = int(input("Masukkan angka bitwise-1: "))
angka2_3011 = int(input("masukkan angka bitwise-2: "))

print("\n angka dalam bentuk desimal dan biner")
print("angka1 =", angka1_3011, "| biner = ", bin(angka1_3011))
print("angka2 =", angka2_3011, "| biner = ", bin(angka2_3011))

# Bitwise AND
hasil_3011 = angka1_3011 & angka2_3011
print("\n Bitwise AND (&)")
print(angka1_3011, "&", angka2_3011, "=", hasil_3011)
print("Biner Hasil =", bin (hasil_3011))
print("Biner Hasil (8 bit) =", format(hasil_3011, "08b"))

# Bitwise OR
hasil_3011 = angka1_3011 | angka2_3011
print("\n Bitwise OR (|)")
print(angka1_3011, "|", angka2_3011, "=", hasil_3011)
print("Biner Hasil =", bin(hasil_3011))
print("Biner Hasil (8 bit) =", format(hasil_3011, "08b"))

# Bitwise XOR
hasil_3011 = angka1_3011 ^ angka2_3011
print("\n Bitwise XOR (^)")
print(angka1_3011, "^", angka2_3011, "=", hasil_3011)
print("Biner Hasil", bin(hasil_3011))
print("Biner Hasil (8 bit) =", format(hasil_3011, "08b"))

# Bitwise NOT
hasil_3011 = ~angka1_3011
print("\n Bitwise NOT (~)")
print("~", angka1_3011, "=", hasil_3011)
print("Biner Hasil", bin(hasil_3011))
print("Biner Hasil (8 bit) =", format(hasil_3011, "08b"))

# Bitwise geser kiri
jumlah_geser = int(input("\n masukkan jumlah pergeseran bit: "))

hasil_3011 = angka1_3011 << jumlah_geser
print("\n Bitwise geser kiri (<<)")
print(angka1_3011, "<<", jumlah_geser, "=", hasil_3011)
print("Biner Hasil", bin(hasil_3011))
print("Biner Hasil (8 bit) =", format(hasil_3011, "08b"))

# Bitwise geser kanan
hasil_3011 = angka1_3011 >> jumlah_geser
print("\n Bitwise geser kanan (>>)")
print(angka1_3011, ">>", jumlah_geser, "=", hasil_3011)
print("Biner Hasil", bin(hasil_3011))
print("Biner Hasil (8 bit) =", format(hasil_3011, "08b"))



3. Operator Bitwise

 angka dalam bentuk desimal dan biner
angka1 = 4 | biner =  0b100
angka2 = 8 | biner =  0b1000

 Bitwise AND (&)
4 & 8 = 0
Biner Hasil = 0b0
Biner Hasil (8 bit) = 00000000

 Bitwise OR (|)
4 | 8 = 12
Biner Hasil = 0b1100
Biner Hasil (8 bit) = 00001100

 Bitwise XOR (^)
4 ^ 8 = 12
Biner Hasil 0b1100
Biner Hasil (8 bit) = 00001100

 Bitwise NOT (~)
~ 4 = -5
Biner Hasil -0b101
Biner Hasil (8 bit) = -0000101

 Bitwise geser kiri (<<)
4 << 3 = 32
Biner Hasil 0b100000
Biner Hasil (8 bit) = 00100000

 Bitwise geser kanan (>>)
4 >> 3 = 0
Biner Hasil 0b0
Biner Hasil (8 bit) = 00000000


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Bentuk Desimal dan Biner<br>
Angka pertama yang digunakan adalah 4, sedangkan angka kedua adalah 8.<br>
Dalam bentuk biner:<br>
4 = 100<br>
8 = 1000<br>
Bentuk biner ini digunakan karena operator bitwise bekerja pada setiap bit dari bilangan tersebut.<br>
2. Bitwise AND (&)<br>
Operasi AND membandingkan setiap bit dari kedua angka. Hasilnya bernilai 1 hanya jika kedua bit yang dibandingkan sama-sama 1.<br>
  0100<br>
& 1000<br>
  0000<br>
Sehingga:<br>
4 & 8 = 0<br>
Hasil dalam 8 bit adalah 00000000.<br>
3. Bitwise OR (|)<br>
Operasi OR menghasilkan nilai 1 apabila salah satu atau kedua bit bernilai 1.<br>
  0100<br>
| 1000<br>
  1100<br>
Biner 1100 memiliki nilai desimal 12.<br>
Sehingga:<br>
4 | 8 = 12<br>
Hasil dalam 8 bit adalah 00001100.<br>
4. Bitwise XOR (^)<br>
Operasi XOR menghasilkan nilai 1 apabila kedua bit berbeda.<br>
  0100<br>
^ 1000<br>
  1100<br>
Karena posisi bit 1 pada angka 4 dan angka 8 berbeda, hasilnya adalah 1100, yaitu 12 dalam desimal.<br>
Sehingga:<br>
4 ^ 8 = 12<br>
Hasil dalam 8 bit adalah 00001100.<br>
5. Bitwise NOT (~)<br>
Operator NOT digunakan untuk membalik bit. Pada Python, operasi NOT menggunakan representasi bilangan bertanda sehingga hasil dari ~4 adalah -5.<br>
Secara matematis, operasi NOT pada bilangan 4 dapat dipahami dengan rumus:<br>
~n = -(n + 1)<br>
Maka:<br>
~4 = -(4 + 1) = -5<br>
Oleh karena hasilnya negatif, tampilan biner dengan format "08b" juga akan menampilkan tanda negatif, sehingga outputnya adalah:<br>
-00000101<br>
6. Bitwise Geser Kiri (<<)<br>
Operator geser kiri digunakan untuk menggeser bit ke sebelah kiri sebanyak jumlah pergeseran yang ditentukan.<br>
Jumlah pergeseran adalah 3.<br>
4 = 00000100<br>
     ↓ geser kiri 3 bit<br>
   00100000<br>
Hasilnya adalah 32.<br>
Sehingga:<br>
4 << 3 = 32<br>
Setiap pergeseran ke kiri satu posisi secara umum setara dengan mengalikan bilangan dengan 2. Karena digeser 3 kali:<br>
4 × 2³ = 4 × 8 = 32<br>
Hasil biner 8 bit adalah 00100000.<br>
7. Bitwise Geser Kanan (>>)<br>
Operator geser kanan digunakan untuk menggeser bit ke sebelah kanan sebanyak 3 posisi.<br>
4 = 00000100<br>
     ↓ geser kanan 3 bit<br>
   00000000<br>
Karena angka 4 digeser ke kanan sebanyak 3 posisi, semua bit 1 akan keluar dari posisi yang digunakan.<br>
Sehingga:<br>
4 >> 3 = 0<br>
Hasil biner 8 bit adalah 00000000.<br>

### 2.2.4 Program Operator Keanggotaan_2611533011
</div>

In [7]:
print("=========================")
print(" 1. OPERATOR KEANGGOTAAN")
print("=========================")

# input beberapa data yang dipisahkan dengan koma
input_data_3011 = input("masukkan beberapa angka, pisahkan dengan koma: ")

# mengubah input menjadi list integer
data_3011 = [int(angka.strip()) for angka in input_data_3011.split(",")]

nilai_dicari_3011 = int(input("masukkan angka yang ingin di cari: "))

#operator in 
hasil_3011 = nilai_dicari_3011 in data_3011 
print("\n Operator keanggotaan IN")
print(nilai_dicari_3011, "in", data_3011, "=", hasil_3011)

# operator not in 
hasil_3011 = nilai_dicari_3011 not in data_3011 
print("\n Operator keanggotaan NOT IN")
print(nilai_dicari_3011, "not in", data_3011, "=", hasil_3011)

 1. OPERATOR KEANGGOTAAN

 Operator keanggotaan IN
9 in [8, 7] = False

 Operator keanggotaan NOT IN
9 not in [8, 7] = True


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data<br>
Pada program, pengguna memasukkan beberapa angka yang dipisahkan dengan tanda koma. Dalam percobaan ini, data yang dimasukkan adalah 8 dan 7.<br>
Setelah diproses oleh program, data tersebut menjadi sebuah list yang berisi:<br>
[8, 7]<br>
Kemudian pengguna memasukkan nilai yang ingin dicari, yaitu 9.<br>
2. Operator Keanggotaan in<br>
Operator in digunakan untuk memeriksa apakah suatu nilai terdapat di dalam kumpulan data.<br>
Pada program, nilai yang dicari adalah 9, sedangkan data yang tersedia adalah [8, 7].<br>
Karena angka 9 tidak terdapat dalam data tersebut, maka hasil pemeriksaannya adalah:<br>
9 in [8, 7] = False<br>
Artinya, nilai 9 tidak menjadi anggota atau tidak ditemukan di dalam list [8, 7].<br>
3. Operator Keanggotaan not in<br>
Operator not in digunakan untuk memeriksa apakah suatu nilai tidak terdapat di dalam kumpulan data.<br>
Karena nilai 9 memang tidak terdapat dalam list [8, 7], maka kondisi not in bernilai:<br>
9 not in [8, 7] = True<br>
Artinya, angka 9 benar-benar tidak terdapat dalam data [8, 7].<br>

### 2.2.5 Program Operator Identitas_2611533011
</div>

In [11]:
print("\n======================")
print("2. OPERATOR IDENTITAS")
print("========================")

# objek1 menggunakan list dari input pengguna 
objek1_3011 = data_3011

# objek2 merujuk pada objek yang sama dengan objek1
objek2_3011 = objek1_3011 

# objek3 memiliki isi sama, tetapi merupakan objek baru
objek3_3011 = data_3011.copy()

print("objek1_3011 =", objek1_3011)
print("objek2_3011 =", objek2_3011)
print("objek3_3011 =",objek3_3011)

# operator is
hasil_3011= objek1_3011 is objek2_3011
print("\n Operator identitas IS")
print("objek1_3011 is objek2_3011 =", hasil_3011)

# operator is not
hasil_3011 = objek1_3011 is objek3_3011
print("\n Operator identitas IS NOT")
print("objek1_3011 is not objek3_3011 =", hasil_3011)

# membandingkan identitas dan nilai 
print("\n membandingkan identitas dan nilai")
print("objek1_3011 is objek3_3011: ", objek1_3011 is objek3_3011)
print("objek1_3011 == objek3_3011: ", objek1_3011 == objek3_3011)


2. OPERATOR IDENTITAS
objek1_3011 = [8, 7]
objek2_3011 = [8, 7]
objek3_3011 = [8, 7]

 Operator identitas IS
objek1_3011 is objek2_3011 = True

 Operator identitas IS NOT
objek1_3011 is not objek3_3011 = False

 membandingkan identitas dan nilai
objek1_3011 is objek3_3011:  False
objek1_3011 == objek3_3011:  True


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Data Objek<br>
Data yang digunakan adalah 8 dan 7, sehingga data_3011 memiliki nilai [8, 7]. Kemudian data tersebut digunakan untuk membuat objek1_3011, objek2_3011, dan objek3_3011.<br>
2. Objek 1 dan Objek 2<br>
objek2_3011 merujuk langsung pada objek1_3011. Artinya, kedua variabel tersebut menunjuk pada objek yang sama. Oleh karena itu, operator is menghasilkan True.<br>
3. Objek 3<br>
objek3_3011 dibuat menggunakan proses copy(). Walaupun memiliki isi yang sama dengan objek1_3011, objek tersebut merupakan objek baru. Oleh sebab itu, identitas keduanya berbeda.<br>
4. Operator is<br>
Perbandingan objek1_3011 is objek2_3011 menghasilkan True karena kedua variabel merujuk pada objek yang sama.<br>
5. Operator is not<br>
Pada program terdapat perbandingan identitas antara objek1_3011 dan objek3_3011. Karena keduanya merupakan objek yang berbeda, maka kondisi is not bernilai True jika ditulis sesuai operator tersebut.<br>
Namun, pada kode yang diberikan, baris:<br>
objek1_3011 is not objek3_3011<br>
seharusnya menghasilkan True, bukan False. Jadi output yang benar untuk bagian tersebut adalah:<br>
objek1_3011 is not objek3_3011 = True<br>
6. Perbandingan Identitas<br>
Perbandingan objek1_3011 is objek3_3011 menghasilkan False karena kedua objek memiliki identitas yang berbeda, meskipun isinya sama.<br>
7. Perbandingan Nilai<br>
Perbandingan objek1_3011 == objek3_3011 menghasilkan True karena isi kedua objek sama, yaitu [8, 7].<br>
8. Kesimpulan<br>
Berdasarkan hasil program, operator identitas digunakan untuk mengetahui apakah dua variabel mengacu pada objek yang sama. Operator is membandingkan identitas objek, sedangkan == membandingkan nilai atau isi objek. Dalam percobaan ini, objek1_3011 dan objek2_3011 merupakan objek yang sama, sedangkan objek3_3011 merupakan objek baru yang memiliki isi sama.
### 2.2.6 Program Operator Perbandingan_2611533011
</div>

In [12]:
# nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# nilai yang dimasukkan akan dikonversi menjadi tipe data integer
# program operator perbandingan dalam python

angka1_3011 = int(input("input angka-1:"))
angka2_3011 = int(input("input angka-2:"))

# Lebih besar dari
hasil_3011 = angka1_3011 > angka2_3011
print("\n Operator lebih besar dari")
print("angka1 > angka 2 =", hasil_3011)

# Lebih kecil dari
hasil_3011 = angka1_3011 < angka2_3011
print("\n Operator lebih kecil dari")
print("angka1 < angka 2 =", hasil_3011)

# Lebih besar dari atau sama dengan
hasil_3011 = angka1_3011 >= angka2_3011
print("\n Operator lebih besar dari atau sama dengan")
print("angka1 >= angka 2 =", hasil_3011)

# Lebih kecil dari atau sama dengan
hasil_3011 = angka1_3011 <= angka2_3011
print("\n Operator lebih kecil dari atau sama dengan")
print("angka1 <= angka 2 =", hasil_3011)

#  Sama dengan
hasil_3011 = angka1_3011 == angka2_3011
print("\n Operator sama dengan")
print("angka1 == angka 2 =", hasil_3011)

# tidak sama dengan
hasil_3011 = angka1_3011 != angka2_3011
print("\n Operator tidak sama dengan")
print("angka1 != angka 2 =", hasil_3011)

# Tambahan: perbandingan berarti dalam python
hasil_3011 = 0 < angka1_3011 < 100
print("\n Perbandingan berantai")
print("0 < angka1 < 100 =", hasil_3011)

hasil_3011 = 0 < angka2_3011 < 100
print("0 < angka2 < 100 =", hasil_3011)


 Operator lebih besar dari
angka1 > angka 2 = False

 Operator lebih kecil dari
angka1 < angka 2 = True

 Operator lebih besar dari atau sama dengan
angka1 >= angka 2 = False

 Operator lebih kecil dari atau sama dengan
angka1 <= angka 2 = True

 Operator sama dengan
angka1 == angka 2 = False

 Operator tidak sama dengan
angka1 != angka 2 = True

 Perbandingan berantai
0 < angka1 < 100 = True
0 < angka2 < 100 = True


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Operator Lebih Besar Dari (>)<br>
Program membandingkan apakah angka1 lebih besar daripada angka2.<br>
6 > 9 = False<br>
Hasilnya False karena angka 6 lebih kecil daripada angka 9.
2. Operator Lebih Kecil Dari (<)<br>
Program membandingkan apakah angka1 lebih kecil daripada angka2.<br>
6 < 9 = True<br>
Hasilnya True karena angka 6 memang lebih kecil daripada angka 9.
3. Operator Lebih Besar Dari atau Sama Dengan (>=)<br>
Operator ini digunakan untuk memeriksa apakah angka1 lebih besar atau sama dengan angka2.<br>
6 >= 9 = False<br>
Hasilnya False karena angka 6 tidak lebih besar dan juga tidak sama dengan angka 9.
4. Operator Lebih Kecil Dari atau Sama Dengan (<=)<br>
Operator ini digunakan untuk memeriksa apakah angka1 lebih kecil atau sama dengan angka2.<br>
6 <= 9 = True<br>
Hasilnya True karena angka 6 lebih kecil daripada angka 9.
5. Operator Sama Dengan (==)<br>
Operator ini digunakan untuk memeriksa apakah angka1 dan angka2 memiliki nilai yang sama.<br>
6 == 9 = False<br>
Hasilnya False karena nilai 6 tidak sama dengan 9.
6. Operator Tidak Sama Dengan (!=)<br>
Operator ini digunakan untuk memeriksa apakah angka1 dan angka2 memiliki nilai yang berbeda.<br>
6 != 9 = True<br>
Hasilnya True karena angka 6 dan 9 memiliki nilai yang berbeda.<br>
7. Perbandingan Berantai Angka1<br>
Program memeriksa apakah nilai angka1 berada di antara 0 dan 100.<br>
0 < 6 < 100 = True<br>
Hasilnya True karena 6 lebih besar dari 0 dan lebih kecil dari 100.
8. Perbandingan Berantai Angka2<br>
Program juga memeriksa apakah nilai angka2 berada di antara 0 dan 100.<br>
0 < 9 < 100 = True<br>
Hasilnya True karena 9 lebih besar dari 0 dan lebih kecil dari 100.
### 2.2.7 Program Operator Logika_2611533011
</div>

In [2]:
# nama variabel ditambah 4 digit nim terakhir contoh: a1_1234
# program ini menggunakan fungsi input()
# program operator logika dalam python

# Memasukkan nilai boolean
# input tidak peka terhadap huruf besar dan kecil
a1_3011 = input("input nilai booelan-1 (true/false): "). strip(). lower() == "true"
a2_3011 = input("input nilai booelan-2 (true/false): "). strip(). lower() == "true"

print("\nA1 =", a1_3011)
print("A2 =", a2_3011)

# konjungsi: bernilai True jika keduanya True
hasil_3011 = a1_3011 and a2_3011
print("\n konjungsi (AND)")
print("A1 and A2 =", hasil_3011)

# disjungsi: bernilai true jika salah satunya true
hasil_3011 = a1_3011 or a2_3011
print("\n disjungsi (OR)")
print("A1 or A2 = ", hasil_3011)

# negasi A1: membalik nilai A1
hasil_3011 = not a1_3011 
print("\n negasi (NOT)")
print("not A1= ", hasil_3011)

# negasi A2: membalik nilai A2
hasil_3011 = not a2_3011 
print("\n negasi (NOT)")
print("not A2= ", hasil_3011)

# XOR: bernilai true jika kedua nilai berbeda
hasil_3011 = a1_3011 != a2_3011
print("\n disjungsi eksklusif (XOR)")
print("A1 XOR A2 =", hasil_3011)


A1 = False
A2 = False

 konjungsi (AND)
A1 and A2 = False

 disjungsi (OR)
A1 or A2 =  False

 negasi (NOT)
not A1=  True

 negasi (NOT)
not A2=  True

 disjungsi eksklusif (XOR)
A1 XOR A2 = False


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Nilai A1 dan A2<br>
Program meminta input berupa nilai Boolean, yaitu true atau false. Karena input yang diberikan adalah 4 dan 7, keduanya tidak sama dengan teks "true". Oleh karena itu, program mengubah kedua nilai tersebut menjadi:<br>
A1 = False<br>
A2 = False<br>
2. Operator Konjungsi (AND)<br>
Operator and menghasilkan True jika kedua nilai bernilai True. Karena A1 dan A2 sama-sama False, maka:<br>
False and False = False<br>
Jadi, hasilnya adalah False.
3. Operator Disjungsi (OR)<br>
Operator or menghasilkan True jika salah satu atau kedua nilai bernilai True. Karena A1 dan A2 sama-sama False, maka:<br>
False or False = False<br>
Jadi, hasilnya adalah False.<br>
4. Negasi (NOT) A1<br>
Operator not digunakan untuk membalik nilai Boolean. Karena A1 bernilai False, maka:<br>
not False = True<br>
Jadi, hasil negasi A1 adalah True.
5. Negasi (NOT) A2<br>
Sama seperti A1, A2 juga memiliki nilai False. Setelah menggunakan operator not, nilainya menjadi:<br>
not False = True<br>
Jadi, hasil negasi A2 adalah True.
6. Operator XOR<br>
XOR menghasilkan True jika kedua nilai berbeda. Dalam percobaan ini A1 dan A2 sama-sama False, sehingga keduanya tidak berbeda.<br>
False XOR False = False<br>
Jadi, hasil XOR adalah False.

</div>

---
<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

## TUGAS PRAKTIKUM

**Soal :**

<div align="center">
<img src="images/Tugas3_if.png" width="500"><br>
<em>Gambar 2.2 : Soal Tugas Praktikum</em>
</div>

**Kode Jawaban :**

a. Kode Program tugas3_2611533011

In [3]:
print ("\n=== SISTEM TRANSAKSI TOKO ===")

nama_3011 = input("Masukkan Nama Pelanggan: ")
status_3011 = input("Masukkan Status Pelanggan (member/nonmember): ")
total_belanja_3011 = int(input("Masukkan Total Belanja : "))
jumlah_barang_3011 = int(input("Masukkan Jumlah Barang : "))
promo_3011 = input("Masukkan Kode Promo : ")

 #========================================
 #  ======DATA TRANSAKSI======
 # =======================================


print("\n=== DATA TRANSAKSI ===")
print("Nama Pelanggan                     : ",nama_3011)
print("Status Pelanggan (member/nonmember): ", status_3011)
print(f"Total Belanja                      :  Rp{total_belanja_3011}")
print("Jumlah Barang                      : ", jumlah_barang_3011)
print("Kode Promo                         : ", promo_3011)

#=========================================
#=========HASIL VALIDASI================
#=========================================

kode_promo_3011 = ["HIDUP IF", "HIDUP FTI", "JALAN SEHAT"]

syarat_total_3011 = total_belanja_3011 >=200000
syarat_jumlah_barang_3011 = jumlah_barang_3011 >=3
valid_status_3011 = status_3011 == "member"
valid_promo_3011 = promo_3011 in kode_promo_3011 


print("\n=== HASIL VALIDASI ===")
print(f"Belanja >= Rp200000                : {syarat_total_3011}") 
print(f"Jumlah Barang >= 3                 : {syarat_jumlah_barang_3011}")
print(f"Status Member                      : {valid_status_3011}")
print(f"Kode Promo Tersedia                : {valid_promo_3011}")
print(f"Mendapatkan Diskon                 : {syarat_jumlah_barang_3011 or syarat_total_3011}")
print(f"Mendapatkan Promo                  : {valid_promo_3011}")

#=========================================
#  =====HASIL PERHITUNGAN============
#=========================================

print("\n=== HASIL PERHITUNGAN ===")
print(f"Diskon                             : Rp{int(0.15 * total_belanja_3011)}")
print(f"Total Pembayaran                   : Rp{int(total_belanja_3011 - 0.15 * total_belanja_3011)}")
print(f"Rata-rata Harga Barang             : Rp{int((total_belanja_3011 - 0.15 * total_belanja_3011) // jumlah_barang_3011)}")

#=========================================
#=========HAK AKSES PELANGGAN===========
#=========================================
 
print("\n=== HAK AKSES PELANGGAN ===")
print("Kode Hak Akses                     : ...")
print(f"Member Access                      : {status_3011 == "member"}")
print(f"Promo Access                       : {promo_3011 in kode_promo_3011}")
print("Free Shipping Access               : ...")

#=========================================
#=========OPERASI BITWISE===============
#=========================================


print("\n=== OPERASI BITWISE ===")

print("\n=== Kode Status Transaksi ===")

#KODE STATUS
#0001 = member
#0010 = total belanja >= Rp 200000
#0100 = jumlah barang >= 3
#1000 = promo

#menggunakan OR (|)
kode_transaksi_3011 =  int(valid_status_3011) << 0| int(syarat_total_3011) << 1 | int(syarat_jumlah_barang_3011) << 2 | int(valid_promo_3011) << 3
kode_referensi_3011 = int(valid_status_3011) << 0| int(syarat_total_3011) << 1 | int(valid_promo_3011) << 3

print(f"{format(int(valid_status_3011) << 0, "04b")} << 0| {format(int(syarat_total_3011) << 1, "04b")} | {format(int(syarat_jumlah_barang_3011) << 2, "04b")} | {format(int(valid_promo_3011) << 3, "04b")}")
print(f"Kode Biner   : {format(kode_transaksi_3011, "04b")}") 
print(f"Kode Desimal : {kode_transaksi_3011}")

#====================================
#========PEMERIKSAAN STATUS=========
#====================================

print("\n=== Pemeriksaan Status ===")


print("\nCek Member")

#menggunakan and (&)

print(f"{format(kode_transaksi_3011, "04b")} & {format(int(valid_status_3011) << 0, "04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_3011) & int(valid_status_3011) << 0, "04b")}")
print(f"Hasil Desimal : {(kode_transaksi_3011) & int(valid_status_3011) << 0}") 

print("\nCek Promo")

print(f"{format(kode_transaksi_3011, "04b")} & {format(int(valid_promo_3011) << 3, "04b")}")
print(f"Hasil Biner   : {format((kode_transaksi_3011) & int(valid_promo_3011) << 3, "04b")}")
print(f"Hasil Desimal  : {(kode_transaksi_3011) & int(valid_promo_3011) << 3}")

#====================================
#======PERBANDINGAN STATUS==========
#============================================

print("\n=== Perbandingan Status ===")

 # menggunakan XOR (^)

print(f"Kode Transaksi : {format(kode_transaksi_3011, "04b")}")
print(f"Kode Referensi : {format(kode_referensi_3011, "04b")}")
print(f"{format(kode_transaksi_3011, "04b")} ^ {format(kode_referensi_3011, "04b")}")
print(f"Hasil Biner    : {format((kode_transaksi_3011) ^ (kode_referensi_3011))}")
print(f"Hasil Desimal  : {(kode_transaksi_3011) ^ (kode_referensi_3011)}")

#=====================================
#=======SHIFT GESER KIRI=============
#=====================================


print("\n=== Shift ===")

#bitwise kiri/ shift kiri (<<)

print(f"{format(kode_transaksi_3011,"04b")} << 1")
print(f"Hasil Biner   : {format((kode_transaksi_3011) << 1,"04b")}")
print(f"Hasil Desimal : {(kode_transaksi_3011) << 1}") 
print("=== SELESAI ===")


=== SISTEM TRANSAKSI TOKO ===

=== DATA TRANSAKSI ===
Nama Pelanggan                     :  AGUNG FIRMANSYAH
Status Pelanggan (member/nonmember):  member
Total Belanja                      :  Rp600000
Jumlah Barang                      :  9
Kode Promo                         :  HEBAT IF

=== HASIL VALIDASI ===
Belanja >= Rp200000                : True
Jumlah Barang >= 3                 : True
Status Member                      : True
Kode Promo Tersedia                : False
Mendapatkan Diskon                 : True
Mendapatkan Promo                  : False

=== HASIL PERHITUNGAN ===
Diskon                             : Rp90000
Total Pembayaran                   : Rp510000
Rata-rata Harga Barang             : Rp56666

=== HAK AKSES PELANGGAN ===
Kode Hak Akses                     : ...
Member Access                      : True
Promo Access                       : False
Free Shipping Access               : ...

=== OPERASI BITWISE ===

=== Kode Status Transaksi ===
0001 << 0| 0010 | 

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Judul Program<br>
Program ini merupakan Sistem Simulasi Transaksi dan Validasi Akses Toko yang dibuat menggunakan bahasa pemrograman Python.<br>
Program digunakan untuk menerima data pelanggan, memeriksa syarat transaksi, menghitung diskon dan total pembayaran, serta meentukan hak akses pelanggan berdasarkan status dan kode promo.<br>
Selain melakukan perhitungan transaksi, program ini juga menerapkan berbagai jenis operator Python, seperti operator aritmatika, perbandingan, logika, keanggotaan, assignment, dan bitwise.<br>
2. Input Data Pelanggan<br>
Pada awal program, pengguna diminta memasukkan beberapa data pelanggan dan transaksi.<br>
Data yang dimasukkan meliputi nama pelanggan, status pelanggan, total belanja, jumlah barang, dan kode promo.<br>
Data nama pelanggan dan kode promo berupa teks, sedangkan total belanja dan jumlah barang berupa bilangan bulat.<br>
Penggunaan input dari pengguna membuat program bersifat interaktif karena hasil transaksi dapat berubah sesuai dengan data yang dimasukkan.<br>
Nama variabel menggunakan angka 3011 sebagai penyesuaian dengan ketentuan penggunaan empat digit terakhir NIM.<br>
3. Data Transaksi<br>
Setelah pengguna memasukkan seluruh data, program menampilkan kembali data transaksi yang telah dimasukkan.<br>
Bagian ini bertujuan untuk memastikan bahwa informasi pelanggan dan transaksi yang diproses oleh program sesuai dengan input pengguna.<br>
Informasi yang ditampilkan meliputi nama pelanggan, status pelanggan, total belanja, jumlah barang, dan kode promo.<br>
4. Validasi Total Belanja<br>
Program kemudian melakukan pemeriksaan terhadap total belanja pelanggan.<br>
Syarat yang digunakan adalah pelanggan harus memiliki total belanja minimal Rp200.000.<br>
Pemeriksaan ini menggunakan operator perbandingan lebih besar atau sama dengan (>=).<br>
Jika total belanja memenuhi syarat, hasil pemeriksaan adalah True.<br>
Jika total belanja kurang dari Rp200.000, hasil pemeriksaan adalah False.<br>
5. Validasi Jumlah Barang<br>
Program juga memeriksa jumlah barang yang dibeli pelanggan.<br>
Syarat yang digunakan adalah pelanggan membeli minimal 3 barang.<br>
Pemeriksaan tersebut menggunakan operator perbandingan >=.<br>
Jika jumlah barang minimal 3, hasilnya adalah True.<br>
Jika jumlah barang kurang dari 3, hasilnya adalah False.<br>
6. Validasi Status Member<br>
Program memeriksa apakah pelanggan memiliki status sebagai member.<br>
Pemeriksaan dilakukan dengan membandingkan status yang dimasukkan pengguna dengan teks member.<br>
Pada bagian ini digunakan operator perbandingan sama dengan (==).<br>
Jika status pelanggan adalah member, maka hasil validasi adalah True.<br>
Jika status pelanggan adalah nonmember, maka hasil validasi adalah False.<br>
7. Validasi Kode Promo<br>
Program menyediakan beberapa kode promo yang dapat digunakan pelanggan, yaitu HIDUP IF, HIDUP FTI, dan JALAN SEHAT.<br>
Program kemudian memeriksa apakah kode promo yang dimasukkan pengguna terdapat di dalam daftar kode promo tersebut.<br>
Pemeriksaan ini menggunakan operator keanggotaan in.<br>
Jika kode promo terdapat dalam daftar, hasilnya adalah True.<br>
Jika kode promo tidak terdapat dalam daftar, hasilnya adalah False.<br>
8. Menentukan Mendapatkan Diskon<br>
Program menggunakan operator logika or untuk menentukan apakah pelanggan mendapatkan diskon berdasarkan kondisi jumlah barang atau total belanja.<br>
Operator or akan menghasilkan True apabila salah satu atau kedua kondisi bernilai benar.<br>
Dengan demikian, berdasarkan kondisi yang dibuat dalam program, pelanggan mendapatkan status diskon apabila jumlah barang minimal 3 atau total belanja minimal Rp200.000.<br>
Perlu diperhatikan bahwa aturan ini berbeda dengan contoh awal yang menggunakan syarat member dan total belanja secara bersamaan.<br>
9. Menentukan Mendapatkan Promo<br>
Program menentukan apakah pelanggan mendapatkan promo berdasarkan validitas kode promo.<br>
Jika kode promo yang dimasukkan tersedia dalam daftar promo, pelanggan mendapatkan status True pada bagian mendapatkan promo.<br>
Jika kode promo tidak tersedia, hasilnya adalah False.<br>
10. Perhitungan Diskon<br>
Program menghitung diskon sebesar 15% dari total belanja.<br>
Perhitungan tersebut menggunakan operator aritmatika perkalian (*).<br>
Nilai 15% dikalikan dengan total belanja untuk mendapatkan jumlah potongan harga.<br>
Sebagai contoh, apabila total belanja sebesar Rp600.000, maka diskon 15% adalah Rp90.000.<br>
11. Perhitungan Total Pembayaran<br>
Setelah nilai diskon diperoleh, program menghitung total pembayaran yang harus dibayar pelanggan.<br>
Perhitungan dilakukan dengan mengurangi total belanja dengan jumlah diskon.<br>
Pada bagian ini digunakan operator aritmatika pengurangan (-).<br>
Jika total belanja Rp600.000 dan diskon Rp90.000, maka total pembayaran menjadi Rp510.000.<br>
12. Perhitungan Rata-rata Harga Barang<br>
Program juga menghitung rata-rata harga barang setelah diskon.<br>
Perhitungan dilakukan dengan membagi total pembayaran setelah diskon dengan jumlah barang yang dibeli.<br>
Pada bagian ini digunakan operator pembagian bulat (//).<br>
Operator // digunakan agar hasil pembagian berupa bilangan bulat tanpa angka desimal.<br>
Sebagai contoh, jika total pembayaran Rp510.000 dan jumlah barang 4, maka rata-rata harga barang adalah Rp56.666.<br>
13. Hak Akses Pelanggan<br>
Bagian hak akses digunakan untuk menentukan fasilitas yang dapat diperoleh pelanggan berdasarkan kondisi transaksi.<br>
Member Access digunakan untuk menunjukkan apakah pelanggan memiliki status member.<br>
Promo Access digunakan untuk menunjukkan apakah pelanggan menggunakan kode promo yang tersedia.<br>
Sementara itu, bagian Free Shipping Access pada program yang diberikan masih ditampilkan sebagai tanda titik tiga sehingga belum memiliki aturan khusus yang menentukan kapan pelanggan mendapatkan akses tersebut.<br>
14. Kode Status Transaksi<br>
Program kemudian membuat sebuah kode status transaksi menggunakan bilangan biner.<br>
Setiap kondisi transaksi diwakili oleh satu bit tertentu.<br>
Bit pertama digunakan untuk menunjukkan status member.<br>
Bit kedua digunakan untuk menunjukkan apakah total belanja memenuhi minimal Rp200.000.<br>
Bit ketiga digunakan untuk menunjukkan apakah jumlah barang minimal 3.<br>
Bit keempat digunakan untuk menunjukkan apakah kode promo tersedia.<br>
Dengan cara tersebut, beberapa kondisi transaksi dapat digabungkan menjadi satu kode status transaksi.<br>
15. Operator Bitwise OR<br>
Dalam pembentukan kode status transaksi digunakan operator bitwise OR (|).<br>
Operator OR digunakan untuk menggabungkan beberapa nilai bit menjadi satu kode transaksi.<br>
Jika semua kondisi bernilai benar, setiap bit akan bernilai 1 sehingga menghasilkan kode biner 1111.<br>
Kode biner 1111 memiliki nilai desimal 15.<br>
16. Operator Bitwise AND<br>
Program menggunakan operator bitwise AND (&) untuk melakukan pemeriksaan terhadap status tertentu dari kode transaksi.<br>
Pemeriksaan pertama digunakan untuk mengecek status member.<br>
Jika bit member aktif, hasil pemeriksaan akan menunjukkan nilai biner 0001 atau nilai desimal 1.<br>
Pemeriksaan kedua digunakan untuk mengecek status promo.<br>
Jika bit promo aktif, hasilnya adalah 1000 atau nilai desimal 8.<br>
Dengan demikian, operator AND digunakan untuk mengambil atau memeriksa bagian tertentu dari kode status transaksi.<br>
17. Operator Bitwise XOR<br>
Program juga menggunakan operator bitwise XOR (^) untuk membandingkan kode transaksi dengan kode referensi.<br>
XOR akan menghasilkan nilai 0 apabila dua bit yang dibandingkan sama dan menghasilkan nilai 1 apabila kedua bit berbeda.<br>
Hasil XOR dapat digunakan untuk mengetahui bagian kode transaksi yang memiliki perbedaan dengan kode referensi.<br>
Sebagai contoh, kode transaksi 1111 dibandingkan dengan kode referensi 1011 menghasilkan 0100.<br>
Nilai biner 0100 memiliki nilai desimal 4.<br>
18. Operator Shift Kiri<br>
Program menggunakan operator shift kiri (<<) untuk menggeser posisi bit ke sebelah kiri.<br>
Pada contoh, kode transaksi 1111 digeser satu posisi ke kiri sehingga menjadi 11110.<br>
Nilai desimal dari 1111 adalah 15, sedangkan setelah digeser satu posisi ke kiri nilainya menjadi 30.<br>
Secara sederhana, pergeseran satu posisi ke kiri memiliki efek yang sama dengan mengalikan nilai dengan 2.<br>
19. Fungsi Format Biner<br>
Program menggunakan fungsi format untuk menampilkan nilai angka dalam bentuk bilangan biner.<br>
Penggunaan format empat digit biner membuat kode status lebih mudah dibaca dan sesuai dengan konsep empat kondisi yang digunakan dalam transaksi.<br>
Contohnya, angka desimal 15 dapat ditampilkan sebagai 1111 dalam bentuk biner.<br>
</div>


---
<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

Jawaban dan penjelasan dari analisi laporan dari tugas3:

1. Operator aritmatika yang digunakan<br>
Operator aritmatika yang digunakan dalam program adalah perkalian, pengurangan, dan pembagian bulat. Perkalian digunakan untuk menghitung besar diskon dari total belanja. Pengurangan digunakan untuk mendapatkan total pembayaran setelah diskon. Sedangkan pembagian bulat digunakan untuk menghitung rata-rata harga barang berdasarkan total pembayaran dan jumlah barang. Operator tersebut membantu program melakukan perhitungan transaksi secara otomatis.

2. Perbedaan operator and, or, dan not<br>
Operator and digunakan ketika semua kondisi harus terpenuhi agar menghasilkan nilai True. Operator or digunakan ketika cukup salah satu kondisi terpenuhi untuk menghasilkan True. Sedangkan operator not digunakan untuk membalik hasil suatu kondisi, yaitu True menjadi False dan False menjadi True.<br>
Pada program yang dibuat, konsep operator logika dapat digunakan untuk menentukan apakah pelanggan memenuhi beberapa persyaratan sekaligus, misalnya jumlah belanja dan jumlah barang harus memenuhi batas tertentu.

3. Fungsi operator in dan not in pada pemeriksaan kode promo<br>
Operator in digunakan untuk memeriksa apakah kode promo yang dimasukkan pelanggan terdapat dalam daftar kode promo yang tersedia. Jika kode yang dimasukkan terdapat dalam daftar, maka hasil pemeriksaannya adalah True.<br>
Sedangkan not in digunakan untuk memeriksa apakah kode promo tidak terdapat dalam daftar. Jika kode tersebut tidak tersedia, maka hasilnya adalah True.<br>
Dalam program, operator ini digunakan untuk menentukan apakah pelanggan mendapatkan akses terhadap promo yang tersedia.

4. Perbedaan is dan == pada Python<br>
Operator == digunakan untuk membandingkan nilai dari dua data. Sedangkan operator is digunakan untuk memeriksa apakah dua objek merupakan objek yang sama.<br>
Dalam program transaksi, operator == digunakan untuk membandingkan status pelanggan, misalnya untuk mengetahui apakah status pelanggan adalah member<br>
Operator is tidak digunakan karena program hanya membutuhkan perbandingan nilai, bukan pemeriksaan objek yang sama.

5. Penggunaan operator bitwise untuk menentukan hak akses pelanggan<br>
Operator bitwise digunakan untuk mengolah kode hak akses dalam bentuk bilangan biner. Setiap posisi bit dapat mewakili status atau hak akses tertentu, seperti member dan promo.<br>
Operator OR (|) digunakan untuk menggabungkan beberapa kode status menjadi satu kode transaksi. Kemudian operator AND (&) digunakan untuk memeriksa apakah suatu hak akses tertentu aktif.<br>
Selain itu, program menggunakan XOR (^) untuk membandingkan kode transaksi dengan kode referensi. Operator shift kiri (<<) digunakan untuk menggeser posisi bit ke sebelah kiri.<br>
Dengan cara tersebut, hak akses pelanggan dapat direpresentasikan menggunakan kode biner.

6. Alur program dari input hingga menghasilkan total pembayaran dan status promo<br>
Alur program dimulai dengan pelanggan memasukkan nama, status pelanggan, total belanja, jumlah barang, dan kode promo.<br>
Setelah data dimasukkan, program menampilkan kembali data transaksi tersebut. Selanjutnya program melakukan validasi untuk mengetahui apakah total belanja telah mencapai batas minimal, jumlah barang memenuhi syarat, pelanggan merupakan member, dan kode promo tersedia.<br>
Setelah proses validasi selesai, program menghitung diskon dan total pembayaran. Program juga menghitung rata-rata harga barang berdasarkan total pembayaran dan jumlah barang.<br>
Selanjutnya program menentukan hak akses pelanggan, seperti akses member dan akses promo. Terakhir, program melakukan operasi bitwise terhadap kode status transaksi untuk menunjukkan proses pemeriksaan hak akses dalam bentuk bilangan biner.

7. Contoh kondisi True dan False<br>
Contoh kondisi True:<br>
Jika pelanggan memiliki total belanja Rp600.000, jika kondisi belanja minimal Rp200.000 akan menghasilkan True, karena Rp600.000 memenuhi batas minimal yang ditentukan.<br>
Contoh kondisi False:<br>
Jika pelanggan membeli 2 barang, maka kondisi jumlah barang minimal 3 akan menghasilkan False, karena jumlah barang tersebut belum memenuhi persyaratan.<br>
Contoh lainnya, jika pelanggan memasukkan kode promo yang tidak terdapat dalam daftar promo, maka pemeriksaan kode promo dengan operator in akan menghasilkan False.<br>
</div>

---
<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">

<h1 align="center"><b>BAB III<br>KESIMPULAN</b></h1>
<p style="text-indent: 40px; text-align: justify; "> Berdasarkan praktikum yang telah dilakukan, dapat disimpulkan bahwa operator Python merupakan bagian penting dalam pengolahan dan manipulasi data pada program. Berbagai jenis operator, seperti operator aritmatika, perbandingan, logika, assignment, keanggotaan, identitas, dan bitwise memiliki fungsi yang berbeda sesuai dengan kebutuhan dalam pemrograman. Dalam konteks pemrograman Python, konsep ini memanfaatkan berbagai operator untuk melakukan perhitungan, membandingkan nilai, mengolah kondisi logika, memberikan nilai pada variabel, memeriksa keanggotaan dan identitas objek, serta memanipulasi data pada tingkat bit, sehingga proses pengolahan data dan pembuatan logika program dapat dilakukan secara lebih efektif dan terstruktur.
</p>
</div>


---
<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">

<h1 align="center"><b>DAFTAR PUSTAKA</b></h1>

1. Python Software Foundation. (2026). The Python Tutorial. Python Documentation.<br> https://docs.python.org/3/tutorial/
2. GeeksforGeeks. (2026). Python Operators.<br> https://www.geeksforgeeks.org/python/python-operators/
3. GeeksforGeeks. (2026). Python Arithmetic Operators.<br> https://www.geeksforgeeks.org/python/python-arithmetic-operators/
4. GeeksforGeeks. (2026). Python Logical Operators. <br>https://www.geeksforgeeks.org/python/python-logical-operators/
5. GeeksforGeeks. (2025). Comparison Operators in Python.<br> https://www.geeksforgeeks.org/python/relational-operators-in-python/
6. GeeksforGeeks. (2026). Python Bitwise Operators.<br> https://www.geeksforgeeks.org/python/python-bitwise-operators/

</div>